lib/model.js

```diff
@@ -2153,6 +2153,11 @@ Model.findOneAndUpdate = function(conditions, update, options, callback) {
   const mq = new this.Query({}, {}, this, this.collection);
   mq.select(fields);
 
+  const connectionConfig = this.collection.conn.config;
+  if (connectionConfig && typeof connectionConfig.useFindAndModify !== 'undefined') {
```

Comment:
We shouldn't rely on setting options in `Model` like this, because it is perfectly valid to do something like `await Model.findOne({ foo: 'bar' }).updateOne({ bar: 'baz' }).findOneAndUpdate()`. Chaining means we need to be careful and make sure we don't check global options until the last minute. Check how we handle `strictQuery` for an example.